src/SuperDumpService/Services/DumpStorageFilebased.cs

```diff
@@ -34,7 +34,11 @@ public class DumpStorageFilebased : IDumpStorage {
 					// backwards compatibility, when Metadata files did not exist. read full json, then store metadata file
 					await CreateMetainfoForCompat(id);
 				}
-				list.Add(ReadMetainfoFile(metainfoFilename));
+				DumpMetainfo dumpMetainfo = ReadMetainfoFile(metainfoFilename);
+				if (settings.Value.DumpRetentionDays != 0 &&  dumpMetainfo.PlannedDeletionDate == null || dumpMetainfo.PlannedDeletionDate == DateTime.MinValue) {
```

Comment:
dumpMetainfo.PlannedDeletionDate == DateTime.MinValue
IMHO you could (and should) use `default` instead of `DateTime.MinValue`

Reply:
This IF is broken! It's always executed, when the OR condition matches!

I'd propose this:
```
if (settings.Value.IsDumpRetentionEnabled() && (dumpMetainfo.PlannedDeletionDate == null || dumpMetainfo.PlannedDeletionDate == default)) {
```

Reply:
Also, `DateTime` is a struct, so it cannot be null.

Maybe we should think about using a nullable type (`DateTime?`). It would explicitly be null if no deletiondate is set.